### PR TITLE
Simple counter to stop the searcher when max depth is reached

### PR DESCRIPTION
**Pull request by swanandx**
We can set `max_depth` in global config. 
As we are exploring the graph, depth means the number of decodings we have performed so far.

This PR solves issue #37

### CODE REVIEW

File: tests/integration_test.rs [hunk 1]
```diff
@@ -19,3 +19,17 @@ fn test_no_panic_if_empty_string() {
     perform_cracking("", config);
     assert_eq!(true, true);
 }
+
+// To test if max depth limit is working, we set max_depth in the global config
+// Due to config being global, we don't want it to interfere with other tests, or vice-versa
```
Comment (bee-san) [inline tests/integration_test.rs:24]: At Monzo we have a function that "cleans" the program so it goes back to default settings, you call it at the start of every test which you might think is "dirty" (with differing configs)

Comment (swanandx) [reply tests/integration_test.rs]: It isn't not possible to change the values inside OnceCell. If we want to mutate it, we would need to wrap it in a mutex like `OnceCell<Mutex<Config>>`

File: src/searchers/bfs.rs [hunk 2]
```diff
@@ -5,16 +5,21 @@ use crate::{decoders::crack_results::CrackResult, filtration_system::MyResults};
...
 
     let mut exit_result: Option<CrackResult> = None;
 
+    // let max_depth = get_config().max_depth;
+
+    let mut curr_depth: u32 = 1; // as we have input string, so we start from 1
+
     // loop through all of the strings in the vec
-    while !current_strings.is_empty() {
+    while !current_strings.is_empty() && curr_depth <= max_depth {
```
Comment (bee-san) [inline src/searchers/bfs.rs:20]: Suggestion, make a function for `max_depth_checker` and have this:

```rust
fn max_depth_checker(current_depth){
    return curr_depth <= max_depth || !(max_depth < 0)
}
```

That way when max_depth is under 0 (-1) we go infinitely (if I get the logic statements the right way around...)

Comment (swanandx) [reply src/searchers/bfs.rs:20]: What about using more _rusty_ way?
```rust
Some(max_depth) // if there is limit
None // go infinitely
```

Comment (bee-san) [reply src/searchers/bfs.rs:20]: hahahahah sure!!!

Comment (swanandx) [reply src/searchers/bfs.rs]: Done

**Review by bee-san — APPROVED** [review]

**Pull request merged**